extensions/amp-story/1.0/amp-story-page.js

```diff
@@ -378,6 +383,23 @@ export class AmpStoryPage extends AMP.BaseElement {
     }
   }
 
+  /** @private */
+  maybeSetStoryNextUp_() {
+    const autoAdvanceAttr = this.element.getAttribute('auto-advance-after');
+    if (!this.isLastPage_() || autoAdvanceAttr !== null) {
+      return;
+    }
+    const storyNextUpParam = Services.viewerForDoc(this.element).getParam(
+      'storyNextUp'
+    );
+    if (storyNextUpParam !== null) {
```

Comment:
Sorry, it's super nitty but what I meant in my first comment is that we should check for this param first, before we do the `matches()` selector that don't need to be performed in most cases. JS <> DOM operations are usually what slows down the code so a good rule of thumb is to run as little as possible of these, but it's super nitty and doesn't really make any diff.

Reply:
Ahhhh, I see. You made sense, I jsut misinterpreted. But I think we've got it now!